File: leetcode/2006 Count Number of Pairs With Absolute Difference K/solution.java
class Solution {
    public int countKDifference(int[] nums, int k) { // LeetCode第一题两数之和变体；用哈希表+一次遍历！！！
        int n = nums.length;
        Map<Integer, Integer> hashRecoed = new HashMap<>();
        int result = 0;
        for (int i=0; i<n; i++) {
            result += hashRecoed.getOrDefault(nums[i]-k, 0); // 此处由getOrDefault()代替containsKey()更方便简洁！！！
            result += hashRecoed.getOrDefault(nums[i]+k, 0);
            hashRecoed.put(nums[i], hashRecoed.getOrDefault(nums[i], 0) + 1);
        }
        return result;
    }
}

File: leetcode/2006 Count Number of Pairs With Absolute Difference K/solution.go
func countKDifference(nums []int, k int) int { // LeetCode第一题两数之和变体；用哈希表+一次遍历！！！
    result := 0
    hashRecord := make(map[int]int)
    for _, num := range nums {
        result += hashRecord[num+k] + hashRecord[num-k] // 此处由getOrDefault()代替containsKey()更方便简洁！！！
    }
    return result
}